server/authentication_test.go

```diff
@@ -18,35 +18,67 @@
 package server
 
 import (
+	"bytes"
 	"fmt"
+	"io"
 	"net/http"
 	"testing"
 
 	"github.com/cockroachdb/cockroach/base"
 	"github.com/cockroachdb/cockroach/kv"
+	"github.com/cockroachdb/cockroach/proto"
 	"github.com/cockroachdb/cockroach/sql/driver"
 	"github.com/cockroachdb/cockroach/testutils"
 	"github.com/cockroachdb/cockroach/ts"
+	"github.com/cockroachdb/cockroach/util"
 	"github.com/cockroachdb/cockroach/util/leaktest"
+	gogoproto "github.com/gogo/protobuf/proto"
 )
 
-func doHTTPReq(t *testing.T, client *http.Client, method, url string) (*http.Response, error) {
-	req, err := http.NewRequest(method, url, nil)
+func doHTTPReq(t *testing.T, client *http.Client, method, url string, body gogoproto.Message) (*http.Response, error) {
+	var b io.Reader
+	if body != nil {
+		buf, err := gogoproto.Marshal(body)
+		if err != nil {
+			t.Fatal(err)
+		}
+		b = bytes.NewReader(buf)
+	}
+	req, err := http.NewRequest(method, url, b)
 	if err != nil {
 		t.Fatalf("%s %s: error building request: %s", method, url, err)
 	}
+	if b != nil {
+		req.Header.Add(util.ContentTypeHeader, util.ProtoContentType)
```

Comment:
It's accepted form that the content-type should be set when there is a body. When there is not, it doesn't actually hurt to set it, but there's really no reason to.